src/hooks/useUpdateURLQueryParams.ts

```diff
@@ -6,14 +7,26 @@ const { stringifyQuery, parseQueryString } = queryString;
 export const useUpdateURLQueryParams = () => {
   const { replace } = useHistory();
   const { search, pathname } = useLocation();
-
-  return (nextQueryParams: StringMap) =>
-    replace(
-      `${pathname}?${stringifyQuery({
+  const result = useMemo(
+    () => (nextQueryParams: StringMap, deleteKeys?: string[]) => {
+      const joinedParams = {
         ...parseQueryString(search),
         ...nextQueryParams,
-      })}`
-    );
+      };
+
+      // remove any keys included in the delete list
+      const filteredParams = deleteKeys?.length
+        ? Object.entries(joinedParams)
+            .filter((p) => !deleteKeys.includes(p[0]))
+            .reduce((curr, accum) => ({ ...accum, [curr[0]]: curr[1] }), {})
+        : joinedParams;
+
+      replace(`${pathname}?${stringifyQuery(filteredParams)}`);
+    },
+    [replace, search, pathname]
```

Comment:
Can you explain why these changes are necessary? 
This is a pretty core piece of functionality in spruce. So i'm reluctant to make changes that change its behavior unless they are necessary.

Reply:
These changes allow the caller to indicate whether a query param should be remove from the URL completely. Instead we could set the query param value to undefined and achieve the same result.